bundles/org.openhab.binding.bmwconnecteddrive/src/main/java/org/openhab/binding/bmwconnecteddrive/internal/utils/ImageProperties.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bmwconnecteddrive.internal.utils;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link ImageProperties} Properties of current Vehicle Image
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ImageProperties {
+    public static final int RETRY_COUNTER = 5;
+    public int failCounter = 0;
+    public String viewport = Constants.EMPTY;
+    public int size = -1;
+
+    public ImageProperties(String viewport, int size) {
+        this.viewport = viewport;
+        this.size = size;
+    }
+
+    public ImageProperties() {
+    }
+
+    public void failed() {
+        failCounter++;
+    }
+
+    public boolean failLimitReached() {
+        return failCounter > RETRY_COUNTER;
+    }
+
+    @Override
+    public String toString() {
+        return new StringBuilder(viewport).append(Integer.toString(size)).toString();
```

Comment:
The direct use of StringBuilder can get rather verbose. So try to avoid using it when only concatenating a small number of things.
```suggestion
        return viewport + size;
```

Reply:
fixed. direct use of String replaced by '+'-operator in most places where this is equivalent to using the StringBuilder.